api/src/main/java/io/grpc/LoadBalancer.java

```diff
@@ -1059,6 +1059,22 @@ public void refreshNameResolution() {
       throw new UnsupportedOperationException();
     }
 
+    /**
+     * Historically the channel automatically refreshes name resolution if any subchannel
+     * connection is broken. It's transitioning to let load balancers make the decision. To
+     * avoid silent breakages, the channel checks if {@link #refreshNameResolution} is called
+     * by the load balancer. If not, it will do it and log a warning. This will be removed in
+     * the future and load balancers are completely responsible for triggering the refresh.
+     * See <a href="https://github.com/grpc/grpc-java/issues/8088">#8088</a> for the background.
+     *
+     * @deprecated Do NOT use.
```

Comment:
I think the messaging here should be more like "this should rarely be used, but sometimes the address for the subchannel wasn't provided by the name resolver and a refresh needs to be directed somewhere else instead. When the refresh goes some place instead, you can call this method to disable the short-term check to detect LB policies that need to be updated for the new expected behavior." And then we wouldn't mark it as `@Deprecated`. At some point in the future this will become a noop and then we _will_ mark it deprecated become we will eventually remove the method.

Reply:
Sure, updated the doc and removed `@Deprecated`.